src/PyCall.jl

```diff
@@ -440,6 +440,59 @@ end
 #########################################################################
 
 """
+    @with
+
+Mimics a Python 'with' statement. Usage:
+
+@with EXPR as VAR begin
+    BLOCK
+end
+
+or 
+
+@with EXPR begin
+    BLOCK
+end
+"""
+macro with(EXPR,as,VAR,BLOCK)
+    @assert as==:as "usage: @with EXPR as VAR BLOCK."
+    _with(EXPR,VAR,BLOCK)
+end
+macro with(EXPR,BLOCK)
+    _with(EXPR,nothing,BLOCK)
+end
+        
+function _with(EXPR,VAR,BLOCK)        
+    quote
+        mgr = $(esc(EXPR))
+        exit = pybuiltin("type")(mgr)[:__exit__]  # Not calling it yet
+        value = pybuiltin("type")(mgr)[:__enter__](mgr)
+        exc = true
+        try
+            try
+                $(VAR==nothing ? :() : :($(esc(VAR)) = value))
+                $(esc(BLOCK))
+            catch err
+                # The exceptional case is handled here
+                exc = false
+                if !(pybuiltin("bool")(exit(mgr, pyimport(:sys)[:exc_info]()...)))
```

Comment:
I was hoping you could do `@pycall exit(mgr, pyimport(:sys)[:exc_info]()...)::Bool` and it would work, but this expects an integer type.

It would be better if the `Bool` conversion were changed to use `PyObject_IsTrue`, and then you could use `::Bool`.